### PR TITLE
gui: Optimize OverviewPage::updateTransactions()

### PR DESCRIPTION
**Pull request by jamescowens**
For wallets with a very large number of transactions, the filter->setLimit(numItems) is extremely expensive... on the order of a few seconds, because it also resorts the list if sort is defined (and it is).

This simply stores the number of rows required for the overview page in the filter, and only calls setLimit if the number of rows is changed with a resize, etc. beyond a reasonable point (3x on purpose). Instead the ui->listTransactions->setRowHidden function is called in a for loop that dynamically sets the rows hidden that are not displayable between the actual display limit and the setLimit value (which is sized initially 3x the number of rows displayable). 

This eliminates the delay when flipping back and forth between the overview and the send, receive, or transaction views. It also improves the resize of the window. I have picked scaling parameters on the size of getLimit versus the number of rows displayable that should minimize or eliminate GUI freezes under normal window resizing.

The wallet still does a list resort every time a block is received, because the underlying wallet model is updated, so that will still cause a GUI freeze for wallets with a large number of transactions. (Note that I am talking about 40000+ transactions.)

This cannot be fixed without a major overhaul architecturally.

### CODE REVIEW

File: src/qt/overviewpage.cpp [hunk 1]
```diff
@@ -161,16 +161,45 @@ void OverviewPage::showEvent(QShowEvent *event)
...
-        filter->setLimit(numItems);
+        int numItems = getNumTransactionsForView();
+
+        // This is a "stairstep" approach, using x3 to x6 factors to size the getLimit.
+        // Based on testing with a wallet with a large number of transactions (40k+)
+        // Using a factor of three is a good balance between the setRowHidden loop
+        // and the very high expense of the getLimit call, which invalidates the filter
+        // and sort, and implicitly redoes the sort, which can take seconds.
+
+        // Most main window resizes will be done without an actual call to getLimit.
```
Comment (cyrossignol) [inline src/qt/overviewpage.cpp]: Same here.

**Review by cyrossignol — APPROVED** [review]
Looks good. Neat way to improve this!

File: src/qt/overviewpage.cpp [hunk 2]
```diff
@@ -161,16 +161,45 @@ void OverviewPage::showEvent(QShowEvent *event)
...
-        const size_t itemHeight = DECORATION_SIZE + ui->listTransactions->spacing();
-        const size_t contentsHeight = ui->listTransactions->height();
-        const size_t numItems = contentsHeight / itemHeight;
-        filter->setLimit(numItems);
+        int numItems = getNumTransactionsForView();
+
+        // This is a "stairstep" approach, using x3 to x6 factors to size the getLimit.
+        // Based on testing with a wallet with a large number of transactions (40k+)
+        // Using a factor of three is a good balance between the setRowHidden loop
+        // and the very high expense of the getLimit call, which invalidates the filter
```
[the remark this answers is not in the document]
Comment (jamescowens) [reply src/qt/overviewpage.cpp:185]: Yes! I will put a commit in to fix.

File: src/qt/overviewpage.cpp (hunk 1 above)
Comment (jamescowens) [reply src/qt/overviewpage.cpp]: Yep.

**Pull request merged**